src/stores/useStore.js

```diff
@@ -0,0 +1,106 @@
+};
+
+const useStore = create( set => ( {
+  cameraPreviewUris: [],
+  cameraRollUris: [],
+  comment: "",
+  currentObservation: {},
+  currentObservationIndex: 0,
+  evidenceToAdd: [],
+  galleryUris: [],
+  groupedPhotos: [],
+  loading: false,
+  observations: [],
+  originalCameraUrisMap: {},
+  photoEvidenceUris: [],
+  savingPhoto: false,
+  unsavedChanges: false,
+  deletePhotoFromObservation: uri => set( state => ( {
+    photoEvidenceUris: [...removePhotoFromList( state.photoEvidenceUris, uri )],
+    cameraPreviewUris: [...removePhotoFromList( state.cameraPreviewUris, uri )],
+    evidenceToAdd: [...removePhotoFromList( state.evidenceToAdd, uri )],
+    observations: removeObsPhotoFromObservation(
+      state.observations[state.currentObservationIndex],
+      uri
+    )
+  } ) ),
+  resetStore: ( ) => set( {
+    cameraPreviewUris: [],
+    cameraRollUris: [],
+    comment: "",
+    currentObservation: {},
+    currentObservationIndex: 0,
+    evidenceToAdd: [],
+    galleryUris: [],
+    groupedPhotos: [],
+    loading: false,
+    observations: [],
+    originalCameraUrisMap: {},
+    photoEvidenceUris: [],
+    savingPhoto: false,
+    unsavedChanges: false
+  } ),
+  setCameraRollUris: uris => set( {
+    cameraRollUris: uris,
+    savingPhoto: false
+  } ),
+  setCameraState: options => set( state => ( {
+    evidenceToAdd: options?.evidenceToAdd || state.evidenceToAdd,
+    cameraPreviewUris: options?.cameraPreviewUris || state.cameraPreviewUris,
+    savingPhoto: options?.evidenceToAdd?.length > 0 || state.savingPhoto,
+    originalCameraUrisMap: options?.originalCameraUrisMap || state.originalCameraUrisMap
+  } ) ),
+  setCurrentObservationIndex: index => set( state => ( {
+    currentObservationIndex: index,
+    currentObservation: state.observations[index]
+  } ) ),
+  setGroupedPhotos: photos => set( {
+    groupedPhotos: photos
+  } ),
+  setObservations: updatedObservations => set( state => ( {
+    observations: updatedObservations,
+    currentObservation: updatedObservations[state.currentObservationIndex]
+  } ) ),
+  setPhotoEvidenceUris: uris => set( {
+    photoEvidenceUris: uris
+  } ),
+  setPhotoImporterState: options => set( state => ( {
+    galleryUris: options?.galleryUris || state.galleryUris,
+    savingPhoto: options?.savingPhoto || state.savingPhoto,
+    evidenceToAdd: options?.evidenceToAdd || state.evidenceToAdd,
+    groupedPhotos: options?.groupedPhotos || state.groupedPhotos,
+    observations: options?.observations || state.observations,
+    currentObservation: options?.observations?.[state.currentObservationIndex]
+    || state.observations?.[state.currentObservationIndex]
+  } ) ),
+  updateComment: newComment => set( { comment: newComment } ),
+  updateObservations: updatedObservations => set( state => ( {
+    observations: updatedObservations,
+    currentObservation: updatedObservations[state.currentObservationIndex],
+    unsavedChanges: true
+  } ) )
+} ) );
+
+export default useStore;
```

Comment:
Even at this early stage, this store is becoming big and complicated. Have you thought about using [slices](https://docs.pmnd.rs/zustand/guides/slices-pattern)?